### ISSUE TITLE
bug: 419 with session mode

### ISSUE THREAD

**Issue by devnote-dev** [TekExplorer/dartactyl#2]
When connecting using the session mode, a 419 error response is returned. This occurs with and without reCAPTCHA enabled. As pointed out in the Discord server, it may be due to a missing or invalid CSRF token.

### Reproducing
```dart
import 'package:dartactyl/dartactyl.dart';

void main() async {
  final client = PteroClient.generate(url: 'https://panel.domain');
  final credentials =
      PteroLoginRequest(username: 'yourUsername', password: 'your_password');
  await client.login(credentials);

  final servers = await client.listServers().then((res) => res.data);
  servers.forEach(
      (s) => print('${s.attributes.identifier}: ${s.attributes.name}'));
}
```

**Comment by TekExplorer** [comment]
The issue seems to be exactly that. Comparing the request/response with browser dev tools, I notice that there is a `set-cookie: XSRF-TOKEN...`

However, for some reason, that same cookie never applies to the client.

Needs investigation.

**Comment by TekExplorer** [comment]
Interestingly, the XSRF token shows up if I do a raw dio.get()

Where in the setup is it losing that?

**Comment by devnote-dev** [comment]
Implementing a GET request [here](https://github.com/TekExplorer/dartactyl/blob/master/lib/src/client.g.dart#L25-L29) to get the token before the POST request should solve it.

**Comment by TekExplorer** [comment]
> Implementing a GET request [here](https://github.com/TekExplorer/dartactyl/blob/master/lib/src/client.g.dart#L25-L29) to get the token before the POST request should solve it.

problem is:
1. its a generated file - and an interceptor is better for that
2. we *never get the token*

The real question is, where are we losing it?

**Comment by devnote-dev** [comment]
I did some testing and the token is returned from `GET /` under the `set-cookie` header, but as you said an interceptor would be needed in order to retrieve that.

**Comment by TekExplorer** [comment]
> I did some testing and the token is returned from `GET /` under the `set-cookie` header, but as you said an interceptor would be needed in order to retrieve that.

Yep, I discovered that too, but I'm not sure how to implement it...

**Comment by TekExplorer** [comment]
I FIGURED IT OUT

I was doing it right for ages, but I just needed to URL decode the token!

I'm getting a 400 instead of a 419 now! Hell Yeah!

**Comment by TekExplorer** [comment]
Fixed with 5b67f0b
